package/network/services/dnsmasq/patches/200-ubus_dns.patch

```diff
@@ -1,16 +1,16 @@
 --- a/src/dnsmasq.c
 +++ b/src/dnsmasq.c
-@@ -2021,6 +2021,10 @@ static void check_dns_listeners(time_t n
- 		  daemon->pipe_to_parent = pipefd[1];
- 		}
- 
+@@ -2213,6 +2213,10 @@
+   status = tcp_from_udp(now, status, header, plen, class, daemon->namebuff, daemon->keyname, 
+ 			server, keycount, validatecount);
+   
 +#ifdef HAVE_UBUS
-+	      drop_ubus_listeners();
++  drop_ubus_listeners();
 +#endif
-+
- 	      /* start with no upstream connections. */
- 	      for (s = daemon->servers; s; s = s->next)
- 		 s->tcpfd = -1; 
++ 
+   /* close upstream connections. */
+   for (s = daemon->servers; s; s = s->next)
+     if (s->tcpfd != -1)
```

Comment:
This change looks wrong to me.
This commit removed the `start with no upstream connections.` block. The place where you added it again is not in the close path and is only called when dnssec is activated, see this commit:
https://thekelleys.org.uk/gitweb/?p=dnsmasq.git;a=commitdiff;h=f5cdb007d8845dde8e5053229f47b46b1b756473

Reply:
@hauke thank you . Corrected with the aid of the commit you mentioned.